### PR TITLE
Fixing conformance tests to pass in register name as a new parameter

### PR DESCRIPTION
**Pull request by sophiebrown01**
Register name is no longer in url, so it cannot be scraped from there; instead we're adding a new parameter and passing that in when running tests from openregister. 

Apologies for the files with no changes; they were modified automatically and included in my first local commit, after which I reverted their changes.

Some refactoring can be done with using 'register' instead of assigning to a new variable 'register_name', but will do this in a separate commit.

### CODE REVIEW

File: openregister-conformance [hunk 1]
```diff
@@ -10,7 +10,9 @@ import pytest
 
 parser = argparse.ArgumentParser()
 parser.add_argument("endpoint", nargs="+", help="URL(s) of register to test")
```
Comment (philandstuff) [inline openregister-conformance]: do we want to (either in this PR or in another) remove the `nargs="+"` bit that allows targetting multiple endpoints from one call? We don't really use this feature anyway.

**Review by philandstuff — COMMENTED** [review]
looks good otherwise, happy to merge once you've responded

File: openregister-conformance (hunk 1 above)
Comment (sophiebrown01) [reply openregister-conformance]: Removed nargs from the 'endpoint' parameter.

File: tests/conftest.py [hunk 2]
```diff
@@ -7,12 +7,17 @@ def pytest_addoption(parser):
     parser.addoption("--endpoint", action="append",
                      help="register endpoints to test")
 
+    parser.addoption("--register", action="append", help="Name of register")
```
Comment (philandstuff) [inline tests/conftest.py:10]: I think we don't want `action="append"` here as that allows multiple arguments

File: openregister-conformance [hunk 3]
```diff
@@ -20,8 +22,10 @@ pytest_args = [base_dir, '-q']
 if args.no_https:
     pytest_args += ['-m','not https']
 
-for register in args.endpoint:
-    print(register)
-    pytest_args += ['--endpoint', register]
+for register_base in args.endpoint:
```
Comment (philandstuff) [inline openregister-conformance]: this for loop doesn't need to be a for loop any more if `args.endpoint` no longer allows a list of arguments

File: tests/conftest.py (hunk 2 above)
Comment (sophiebrown01) [reply tests/conftest.py:10]: Fixed

File: openregister-conformance (hunk 3 above)
Comment (sophiebrown01) [reply openregister-conformance]: Fixed

File: tests/conftest.py [hunk 4]
```diff
@@ -7,7 +7,7 @@ def pytest_addoption(parser):
     parser.addoption("--endpoint", action="append",
                      help="register endpoints to test")
 
-    parser.addoption("--register", action="append", help="Name of register")
+    parser.addoption("--register", help="Name of register")
```
Comment (philandstuff) [inline tests/conftest.py]: you missed the one above for `--endpoint` :smile:

Comment (philandstuff) [reply tests/conftest.py]: argh, I think removing `action="append"` has broken stuff, sorry! :confused:

File: tests/conftest.py (hunk 2 above)
Comment (sophiebrown01) [reply tests/conftest.py:10]: Added back in.

**Pull request merged**